nucypher/blockchain/eth/interfaces.py

```diff
@@ -234,9 +235,10 @@ def add_provider(self,
                     from nucypher.utilities.sandbox.constants import PYEVM_GAS_LIMIT
                     genesis_params = PyEVMBackend._generate_genesis_params(overrides={'gas_limit': PYEVM_GAS_LIMIT})
                     pyevm_backend = PyEVMBackend(genesis_parameters=genesis_params)
+                    pyevm_backend.reset_to_genesis(genesis_params=genesis_params,
```

Comment:
:+1: - Excellent to see this come around full-circle

Reply:
Also - As a future optimization, this method might be a way to speed up tests by avoiding airdrops and bypassing contract deployment.

Reply:
Perhaps something we should describe in an issue or TODO?